site/docs/v1/tech/admin-guide/configuration-management/terraform.adoc

```diff
@@ -0,0 +1,502 @@
+  first_name               = "John"
+  last_name                = "Doe"
+  middle_name              = "William"
+  password_change_required = true
+  password                 = "@CfosPAVT3&hCzz5c^&#2F5BxNUY$X!@s!7Wx9bd6Yon54e3"
+  username_status          = "ACTIVE"
+}
+
+resource "fusionauth_registration" "forum-admin1-admin-role" {
+  user_id        = fusionauth_user.forum-admin1.id
+  application_id = fusionauth_application.forum.id
+  roles          = ["admin"]
+}
+
+resource "fusionauth_registration" "forum-user1-user-role" {
+  user_id        = fusionauth_user.forum-user1.id
+  application_id = fusionauth_application.forum.id
+  roles          = ["user"]
+}
+----
+
+Once you are happy with your configuration run `terraform plan` and if you are not experiencing errors and are ok with the planned changes you can go ahead with `terraform apply`.
+
+== Terraform Update
+
+Once a resource is either created or imported you can change your main.tf according to the documentation.
+
+* link:https://registry.terraform.io/providers/gpsinsight/fusionauth/latest/docs/resources/tenant[Terraform Resource: Tenant]
+* link:https://registry.terraform.io/providers/gpsinsight/fusionauth/latest/docs/resources/application[Terraform Resource: Application]
+* link:https://registry.terraform.io/providers/gpsinsight/fusionauth/latest/docs/resources/application_role[Terraform Resource: Application Role]
+* link:https://registry.terraform.io/providers/gpsinsight/fusionauth/latest/docs/resources/user[Terraform Resource: User]
+* link:https://registry.terraform.io/providers/gpsinsight/fusionauth/latest/docs/resources/registration[Terraform Resource: Registration]
+
+After you are done with your configuration changes, run `terraform plan` to check the planned changes and review potential errors. After that you can go ahead with `terraform apply`.
+
+If you want to know what already has been defined by Terraform but not specified in your `.tf` files you can run `terraform show`.
+
+The configuration can get very large and if you want to show only very specific resources you can list all resources with `terraform state list` and show the resource state with `terraform state show <resource-name>` accordingly.
+
+=== Ignore Lifecycle Changes
+//source: ignore lifecycle changes https://itnext.io/how-and-when-to-ignore-lifecycle-changes-in-terraform-ed5bfb46e7ae
+Every time you plan new changes you could face the challenge that information has been updated in your FusionAuth installation. You can either align your configuration with the installation or decide to link:https://developer.hashicorp.com/terraform/language/meta-arguments/lifecycle#ignore_changes[ignore_changes] in your configuration.
+
+There are two very common examples in FusionAuth resources. One if you manage users through Terraform, where you during creation set password_change_required to true. Or where the user starts changing personal information through FusionAuth or Application.
```

Comment:
I think this is a valid reason not to use terraform to manage users :) . I think you could remove this section if we changed the example to not manage users/registrations.